lib/watir/element_collection.rb

```diff
@@ -162,6 +163,23 @@ def elements
       end
     end
 
+    def elements_with_tags
+      els = elements
+      if @selector[:tag_name]
+        els.map { |e| [e, @selector[:tag_name]] }
+      else
+        retries = 0
+        begin
+          els.zip(browser.wd.execute_script('return arguments[0].map(function(e) {return e.localName});', els))
```

Comment:
This shouldn't have `#wd`

Reply:
It has to or it will add an extra wait which is unnecessary

Reply:
Or rather, it calls `wrap_elements_in` which defeats the purpose of the PR

Reply:
It shouldn't, it's passing in a Selenium element not a Watir element...

Reply:
ohh, wrap, right...

Reply:
You're right, it doesn't return elements anymore; just the `tag_name` so it won't wrap anything. Should be fine to remove it.